From 6d7f5d443187760798fbb8dc5a50b77ce256c53e Mon Sep 17 00:00:00 2001
From: Ryan Lortie <desrt@desrt.ca>
Date: Mon, 20 Oct 2014 17:16:22 -0400
Subject: [PATCH] wayland: don't init if XDG_RUNTIME_DIR missing

wayland doesn't strictly follow the XDG_RUNTIME_DIR spec by falling back
to another directory in case the runtime dir is not properly set.

When this variable is unset, wayland will log an error to us, which we
treat as fatal, aborting the entire program.

Skip ourselves all the trouble and don't try to bring up the wayland
backend when we know it will fail in this way.

https://bugzilla.gnome.org/show_bug.cgi?id=738873
---
 gdk/wayland/gdkdisplay-wayland.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/gdk/wayland/gdkdisplay-wayland.c b/gdk/wayland/gdkdisplay-wayland.c
index 5b6fc92..d9fffe4 100644
--- a/gdk/wayland/gdkdisplay-wayland.c
+++ b/gdk/wayland/gdkdisplay-wayland.c
@@ -217,6 +217,13 @@ _gdk_wayland_display_open (const gchar *display_name)
   GdkDisplay *display;
   GdkWaylandDisplay *display_wayland;
 
+  /* If this variable is unset then wayland initialisation will surely
+   * fail, logging a fatal error in the process.  Save ourselves from
+   * that.
+   */
+  if (g_getenv ("XDG_RUNTIME_DIR") == NULL)
+    return NULL;
+
   wl_log_set_handler_client(log_handler);
 
   wl_display = wl_display_connect(display_name);
-- 
2.1.0

